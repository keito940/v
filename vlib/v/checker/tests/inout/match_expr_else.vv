type A = int | string | f64

fn main() {
	x := A('test')
	res := match x {
		int {
			'int'
		}
		string {
			'string'
		}
	}
}
