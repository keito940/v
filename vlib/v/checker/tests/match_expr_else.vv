type A = int | string | f64

fn main() {
	x := A('test')
	_ = match x {
		int {
			'int'
		}
		string {
			'string'
		}
	}
	_ := match x {
		int {
			'int'
		}
		string {
			'string'
		}
		f64 {
			'f64'
		}
		else {
			'else'
		}
	}
	_ := match x {
		int {
			'int'
		}
		string {
			'string'
		}
		else {
			'else'
		}
		f64 {
			'f64'
		}
	}
}
