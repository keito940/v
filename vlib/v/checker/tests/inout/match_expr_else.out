vlib/v/checker/tests/inout/match_expr_else.v:5:9: error: match must be exhaustive (add match branches for: `f64` or an else{} branch)
    3| fn main() {
    4|     x := A('test')
    5|     res := match x {
                  ~~~~~~~~~
    6|         int {
    7|             'int'
